### PR TITLE
 Add proxy support for switch

### PR DESCRIPTION
**Pull request by pengz1**
Prioritize "X-Real-IP" for switch getting profile to support proxy.

### CODE REVIEW

**Comment by pengz1** [comment]
@iceiilin @lanchongyizu @sunnyqianzhang

File: spec/lib/api/2.0/profiles-spec.js [hunk 1]
```diff
@@ -275,6 +275,23 @@ describe('Http.Api.Profiles', function () {
     });
 
     describe("2.0 GET /profiles/switch/:vendor", function() {
+        it.only("should get switch profile via proxy", function() {
```
Comment (yyscamper) [inline spec/lib/api/2.0/profiles-spec.js]: remove `only`!

File: lib/api/2.0/profiles.js [hunk 2]
```diff
@@ -40,8 +40,9 @@ var profilesGet = controller(function (req, res) {
 
 // GET /api/2.0/profiles/switch/:vendor
 var profilesGetSwitchVendor = controller(function(req, res) {
+    var requestIp = req.get("X-Real-IP") || req.connection._peername.address;
```
Comment (lanchongyizu) [inline lib/api/2.0/profiles.js]: @RackHD/corecommitters what is `req.connection._peername.address`?

**Comment by PengTian0** [comment]
test this please

**Comment by yyscamper** [comment]
jenkins: test this please

**Comment by PengTian0** [comment]
test this please

**Comment by PengTian0** [comment]
test this please

**Comment by panpan0000** [comment]
test this please

**Comment by panpan0000** [comment]
test this please

**Comment by sunnyqianzhang** [comment]
:+1:

**Pull request merged**